test/functional/test_framework/wallet.py

```diff
@@ -119,12 +119,19 @@ def _bulk_tx(self, tx, target_weight):
     def get_balance(self):
         return sum(u['value'] for u in self._utxos)
 
-    def rescan_utxos(self):
+    def rescan_utxos(self, *, include_mempool=True):
         """Drop all utxos and rescan the utxo set"""
         self._utxos = []
         res = self._test_node.scantxoutset(action="start", scanobjects=[self.get_descriptor()])
         assert_equal(True, res['success'])
+        spent_txs = set()
+        if not include_mempool:
+            for txid in self._test_node.getrawmempool():
+                tx = self._test_node.decoderawtransaction(self._test_node.getrawtransaction(txid))
+                spent_txs.update([(i["txid"], i["vout"]) for i in tx["vin"]])
         for utxo in res['unspents']:
+            if not include_mempool and (utxo["txid"], utxo["vout"]) in spent_txs:
```

Comment:
Again, this will only *exclude* spent utxos in the mempool, but never *include* created ones, no?

Reply:
Do you mean the utxo created by vout?

Reply:
Yes, a transaction in the mempool may have outputs (vout) that pay to the miniwallet

Reply:
Thank you for your patience, and I'm sorry for the misunderstanding.

Reply:
> Or could I simply assume that all outputs (vout) are paid to the miniwallet?

No, only the ones that pay to the address of the miniwallet.

Reply:
There are so many different types of `scriptPubKey`, do we have a RPC API that can check if a certain `vout` belongs to the miniwallet?  Or should I write "helper method" to do the work?

Reply:
You can use `scan_tx`